validations/src/validations.rs

```diff
@@ -767,6 +759,22 @@ pub fn validate_tally_transaction<'a>(
         num_commits,
     )?;
 
+    if pkh_liars.len() != ta_tx.out_of_consensus.len() {
+        return Err(TransactionError::MismatchingOutOfConsensusNumber.into());
+    }
+
+    let tally_liars: HashSet<_> = ta_tx.out_of_consensus.iter().collect();
```

Comment:
You can simplify this comparison by sorting the two vectors and comparing for equality, you don't need a hashset.